weaver/vault/utils.py

```diff
@@ -0,0 +1,247 @@
+
+
+def get_vault_url(file, container=None):
+    # type: (Union[VaultFile, uuid.UUID, str], Optional[AnySettingsContainer]) -> str
+    """
+    Obtain the vault link corresponding to the file.
+    """
+    if isinstance(file, uuid.UUID) or is_uuid(file):
+        file_id = str(file)
+    else:
+        file_id = file.id
+    settings = get_settings(container)
+    base_url = get_weaver_url(settings)
+    vault_url = base_url + sd.vault_file_service.path.format(file_id=file_id)
+    return vault_url
+
+
+def map_vault_location(reference, container=None, url=False, exists=True):
+    # type: (str, AnySettingsContainer, bool, bool) -> Optional[str]
+    """
+    Convert back and forth between the URL and local path references of the :term:`Vault` file.
+
+    .. seealso::
+        Similar operation to :func:`weaver.wps.utils.map_wps_output_location`.
+
+    .. warning::
+        Does not validate access token to retrieve the file. It is assumed that pre-valuation was accomplished.
+
+    :param reference: Local file path or file URL to be mapped.
+    :param container: Retrieve application settings.
+    :param url: Perform URL mapping (local path -> URL endpoint), or map to local path (URL -> local path).
+    :param exists: Ensure that the mapped file exists, otherwise don't map it (otherwise ``None``).
+    :returns: Mapped reference if applicable, otherwise ``None``.
+    """
+    scheme = urlparse(reference).scheme
+    base = get_vault_dir(container)
+    if url and scheme == "file":
+        reference = reference[7:]
+    if scheme in ["http", "https"]:
+        file_path = sd.vault_file_service.path.format(file_id="")
+        file_id = reference.split(file_path, 1)[-1]
+    elif reference.startswith(base):
+        file_base = base + "/" if not base.endswith("/") else base
+        file_id = reference.split(file_base)[-1].split("/", 1)[0]
+    else:
+        file_id = ""
+    if not file_id:
+        return None
+
+    db = get_db(container)
+    vault = db.get_store(StoreVault)
+    file = vault.get_file(file_id, nothrow=True)
+    href = get_vault_url(file, container)
+    path = get_vault_path(file, container)
+
+    if exists and not os.path.isfile(path):
+        return None
+    return href if url else path
+
+
+def parse_vault_token(header, unique=False):
+    # type: (Optional[str], bool) -> Dict[Optional[str], str]
+    """
+    Parse the authorization header value to retrieve all :term:`Vault` access tokens and optional file UUID.
+
+    .. seealso::
+        - Definition of expected format in :ref:`file_vault_token`.
+        - :class:`sd.XAuthVaultFileHeader`
+
+    :param header: Authorization header to parse.
+    :param unique: Whether only one or multiple tokens must be retrieved.
+    :return: Found mapping of UUID to access token. If unique, UUID can be ``None``.
+    """
+    if not isinstance(header, str):
+        return {}
+    header = header.lower()
+    if unique and "," in header:
+        return {}
```

Comment:
Instead of returning empty mappings, should the code raise errors for an invalid authorization headers?

Reply:
It is left to the calling method to raise the appropriate HTTP code.